packages/ses/src/error/note-log-args.js

```diff
@@ -7,36 +8,37 @@ const { freeze } = Object;
 const { isSafeInteger } = Number;
 
 /**
+ * @template K,V
  * @typedef {object} DoublyLinkedCell
  * A cell of a doubly-linked ring, i.e., a doubly-linked circular list.
  * DoublyLinkedCells are not frozen, and so should be closely encapsulated by
  * any abstraction that uses them.
- * @property {DoublyLinkedCell} next
- * @property {DoublyLinkedCell} prev
- * @property {object} key
- * @property {any} value;
+ * @property {DoublyLinkedCell<K, V>} next
+ * @property {DoublyLinkedCell<K, V>} prev
+ * @property {WeakMap<object, V>} keyValue
  */
 
 /**
  * Makes a new self-linked cell. There are two reasons to do so:
  *    * To make the head sigil of a new initially-empty doubly-linked ring.
  *    * To make a non-sigil cell to be `spliceAfter`ed.
  *
- * @param {object} key
- * @param {any} value
- * @returns {DoublyLinkedCell}
+ * @template K,V
```

Comment:
Might be cleaner to constraint the `K` parameter?

```suggestion
 * @template {object} K
 * @template V
```

Reply:
I'd love to constrain, but this syntax (which is the only one that could conceivably do the right thing) does not do the right thing.  Without `noImplicitAny: true`, it does *not* produce `<K extends object, V>`, it results in just `<K extends unknown, V>` which causes all kinds of issues with trying to use it as `WeakMap<K, V>`.

So, I won't tighten up the types, but I will add a comment to that effect.

Reply:
ah yeah I wish we could switch to  `noImplicitAny`...

Anyway, probably moot given my other suggestion to drop this altogether.